mypy/checker.py

```diff
@@ -2725,26 +2744,22 @@ def flatten(t: Expression) -> List[Expression]:
         return [t]
 
 
-def get_isinstance_type(expr: Expression, type_map: Dict[Expression, Type]) -> Type:
+def get_isinstance_type(expr: Expression, type_map: Dict[Expression, Type]) -> List[TypeRange]:
     all_types = [type_map[e] for e in flatten(expr)]
-
-    types = []  # type: List[Type]
-
+    types = []  # type: List[TypeRange]
     for type in all_types:
         if isinstance(type, FunctionLike):
             if type.is_type_obj():
                 # Type variables may be present -- erase them, which is the best
                 # we can do (outside disallowing them here).
                 type = erase_typevars(type.items()[0].ret_type)
-
-            types.append(type)
-
-    if len(types) == 0:
-        return None
-    elif len(types) == 1:
-        return types[0]
-    else:
-        return UnionType(types)
+            types.append(TypeRange(type, is_upper_bound=False))
+        elif isinstance(type, TypeType):
+            types.append(TypeRange(type.item, is_upper_bound=True))
+        else:  # we didn't see an actual type, but rather a variable whose value is unknown to us
+            return None
+    assert len(types) != 0
```

Comment:
I wonder if we should return `None` if the length if 0? For example, what happens if we have `isinstance(x, ())`?